### ISSUE TITLE
Is it safe to remove redundant threads in JobSystem on running the engine

### ISSUE THREAD

**Issue by FinleyPan** [google/filament#4138]
I found [ctor](https://github.com/google/filament/blob/1bd91e7e945a3b95e8f17ef0cd0c915f22c4192f/libs/utils/src/JobSystem.cpp#L146-L155) of field `mJobSystem` of `FEngine` will launch many theads:
```cpp
...
for (size_t i = 0, n = states.size(); i &lt; n; i++) { 
        auto&amp; state = states[i];
        state.rndGen = default_random_engine(rd());
        state.id = (uint32_t)i;        
        state.js = this;
        if (i &lt; hardwareThreadCount) {           
            // don't start a thread of adoptable thread slots            
            state.thread = std::thread(&JobSystem::loop, this, &state);
        }
}
```
And it seems no workloads dispatched to these threads on running. I need reduce cpu usage as much as possible, and these redundant threads consume cpu a bit according to my profiling:
![image](https://user-images.githubusercontent.com/26212839/122025748-32a41980-cdfc-11eb-856c-34b6a2864ade.png)
Basically, I comment this [line](https://github.com/google/filament/blob/1bd91e7e945a3b95e8f17ef0cd0c915f22c4192f/libs/utils/src/JobSystem.cpp#L153):
```cpp
// state.thread = std::thread(&JobSystem::loop, this, &state);
```
After recompiling anything in my application goes well, but I think a canonical answer from the community is necessary. So is it safe to remove these redundant threads?

**Comment by romainguy** [comment]
You can do that but unused or seldom used threads won't increase CPU usage. That said we were recently looking into this and will lower the number of threads created.
